### ISSUE TITLE
Metrics logged during sanity checking still appears in training

### ISSUE THREAD

**Issue by eladsegal** [PyTorchLightning/pytorch-lightning#8162]
## 🐛 Bug

Metrics logged during sanity checking still appears in training.

## Reproduction using the BoringModel
The code logs in `validation_step` and `validation_epoch_end` only when `self.trainer.sanity_checking` is `True`, but the following progress bar is outputted. This demonstrates leakage of the logging from the sanity checking to the training.
`Epoch 0: 100%
626/626 [00:01<00:00, 546.35it/s, loss=1.54e-14, v_num=2, val_step_metric_epoch=42.00, val_epoch_end_metric=1337.0`

https://colab.research.google.com/drive/1IQSw9AuoK5OH-NYHrk5rhX2Aa7bylSgg?usp=sharing

### Expected behavior
Metrics logged during sanity checking should be cleared or not logged at all.
Progress bar, logger and callback metrics should be empty when the training loop starts.

### Environment
```
* CUDA:
	- GPU:
		- Tesla T4
	- available:         True
	- version:           10.2
* Packages:
	- numpy:             1.19.5
	- pyTorch_debug:     False
	- pyTorch_version:   1.9.0+cu102
	- pytorch-lightning: 1.3.7post0
	- tqdm:              4.41.1
* System:
	- OS:                Linux
	- architecture:
		- 64bit
		- 
	- processor:         x86_64
	- python:            3.7.10
	- version:           #1 SMP Sat Jun 5 09:50:34 PDT 2021
```

**Comment by tchaton** [comment]
Dear @eladsegal,

Yes, sounds like this proposed behaviour might be better.
I believe some users asked to keep the validation in the trainer progress bar as it would disappear too quickly.

@awaelchli @justusschock Any thoughts about this ?

Best,
T.C

**Comment by awaelchli** [comment]
Yes, agreed that it should not output anything, that's the intention. @eladsegal interested looking into this?

**Comment by eladsegal** [comment]
I would, but I see you set it as P0 and since I am not familiar with the codebase I definitely won't have the time to do so soon.

I currently use the following workaround in the validation methods/hooks:
```
if self.trainer.sanity_checking:
    self.log(...)
```

**Comment by tchaton** [comment]
Dear @eladsegal,

I am taking this issue :) I will let you take the next one, promise !

Best,
T.C
